### ISSUE TITLE
Connman systemd unit is incorrect

### ISSUE THREAD

**Issue by Emm** [NixOS/nixpkgs#10314]
Connman doesn't work out of the box. Currently, it appears to start before wpa_supplicant finishes and hence gives a "no carrier" error when scanning for wifi.

It should probably have a "Wants" dependency on wpa_supplicant to ensure proper dependency order.

At the moment, systemctl claims that connman has been active since 2min 13s, but wpa_supplicant since only 41s.

This is on 15.09

**Comment by rnhmjoj** [comment]
Is that it? I set networking.wireless.enable to fix this error.

**Comment by Emm** [comment]
I have that on.

**Comment by fazo96** [comment]
I had the "no carrier" error before but random restarting connman or something always fixed it. However I still have the issue sometimes and always have to restart wpa_supplicant and connman services to solve it. If I have time, I'll research the issue and try your fix to figure it out.

**Comment by Emm** [comment]
Interestingly enough, I don't think it's a race condition between connman and wpa_supplicant...

What I have observed is that (previously), it took some time for /sys/class/net/wlan0 to appear. From what I can deduce of the script run before wpa_supplicant, nixos feeds the content of /sys/class/net to wpa_supplicant, which is bound to be a problem if it's not yet populated.

I wrote "previously", because after upgrading today, the kernel complains about a missing iwlwifi-6000-4.ucode file, and I need to reload the iwlwifi module and restart wpa_supplicant in order to get wifi (this is not progress :( ).
